tests/functions/test_expand_grid.py

```diff
@@ -0,0 +1,635 @@
+            if not isinstance(value.index, pd.MultiIndex):
+                if value.name:
+                    value = value.to_frame(name=f"{key}_{value.name}")
+                    dfs.append(value)
+                else:
+                    value = value.to_frame(name=f"{key}")
+                    dfs.append(value)
+            else:
+                raise ValueError(
+                    "expand_grid does not work with pd.MultiIndex")
+        #process dataframe
+        if isinstance(value, pd.DataFrame):
+            if value.empty:
+                raise ValueError("passed DataFrame cannot be empty")
+            if not (isinstance(value.index, pd.MultiIndex)
+                    or isinstance(value.columns, pd.MultiIndex)):
+                #add key to dataframe columns
+                value = value.add_prefix(f"{key}_")
+                dfs.append(value)
+            else:
+                raise ValueError(
+                    "expand grid does not work with pd.MultiIndex")
+        #process lists
+        if isinstance(value, list):
+            if not value:
+                raise ValueError("passed value cannot be empty")
+            elif np.array(value).ndim == 1:
+                checklist = (type(None), str, int, float, bool)
+                check = (isinstance(internal, checklist) for internal in value)
+                if all(check):
+                    dicts.update({key: value})
+                else:
+                    raise ValueError("values in iterable must be scalar")
+            elif np.array(value).ndim == 2:
+                value = pd.DataFrame(value).add_prefix(f"{key}_")
+                dfs.append(value)
+            else:
+                raise ValueError("sequence's dimension should be 1d or 2d")
+
+    return dfs, dicts
+
+
+#computation for values that are not arrays/dataframes/series
+#these are collected into dictionary and processed with numpy meshgrid
+def grid_computation_dict(dicts):
+    #actual computation
+    if len(dicts) == 1:
+        key = list(dicts.keys())[0]
+        value = list(dicts.values())[0]
+        final = pd.DataFrame(value, columns=[key])
+    else:
+        res = np.meshgrid(*dicts.values())
+        #create structured array
+        #keeps data type of each value in the dict
+        outcome = np.core.records.fromarrays(res, names=",".join(dicts))
+        #reshape into a 1 column array
+        #using the size of any of the arrays obtained from the meshgrid computation
+        outcome = np.reshape(outcome, (np.size(res[0]), 1))
+        #flatten structured array into 1d array
+        outcome = np.concatenate(outcome)
+        #sort array
+        outcome.sort(axis=0, order=list(dicts))
+        #create dataframe
+        final = pd.DataFrame.from_records(outcome)
+    return final
+
+
+#this is for dataframes/series
+#this should be the final output if there are lists or lists and dicts
+#returned from check instance
+def compute_two_dfs(df1, df2):
```

Comment:
I believe the same comment as above applies here.

Copy/pasting it for immediate visible context: This function should be imported from the library, rather than re-defined here.